op-chain-ops/deployer/pipeline/implementations.go

```diff
@@ -93,9 +96,6 @@ func DeployImplementations(ctx context.Context, env *Env, artifactsFS foundry.St
 		DisputeGameFactoryImplAddress:           dio.DisputeGameFactoryImpl,
 		StateDump:                               dump,
 	}
-	if err := env.WriteState(st); err != nil {
```

Comment:
out of interest, why don't you write the state here anymore?

Reply:
It'll help us support writing to an L1 genesis in the future. The caller should decide when it writes state.